### PR TITLE
Graphql 5.3

### PR DESCRIPTION
**Pull request by pleandre**
Upgrading to Graph QL 5.3 to use with the last version of Graph QL Server and dotnet 6.

I still have an issue in the Graph QL Subscription Sample Project if someone can help me have a look.
I couldn't find the issue (it runs but there is an exception on runtime when you update a field and have a subscription), I would need to dig dipper in Graph QL dotnet internals and all the reflection but I didn't have time.

image

Other things seems to work

### CODE REVIEW

**Comment by pestrada-trimble** [comment]
This would also help resolve the [NewtonsoftJson vulnerability for versions < 13.0.1](https://www.mend.io/vulnerability-database/WS-2022-0161)

Currently this package references `GraphQL.NewtonsoftJson/4.1.0` which references `NewtonsoftJson/12.0.3`

**Comment by sungam3r** [comment]
I advise to switch to STJ whenever possible.

File: src/GraphQL.Conventions/Adapters/Resolvers/EventStreamResolver.cs [hunk 1]
```diff
@@ -1,30 +1,26 @@
...
             {
                 result = (result as Task<object>).Result;
             }
             return (IObservable<object>)result;
         }
-
-        public override object Resolve(IResolveFieldContext context)
-        {
-            return context.Source;
-        }
```
Comment (Shane32) [inline src/GraphQL.Conventions/Adapters/Resolvers/EventStreamResolver.cs]: This is critical for subscription support.

File: src/GraphQL.Conventions/Adapters/Resolvers/EventStreamResolver.cs [hunk 2]
```diff
@@ -1,30 +1,26 @@
...
-using GraphQL.Resolvers;
-using GraphQL.Subscription;
+﻿using GraphQL.Resolvers;
 using System;
 using System.Threading.Tasks;
 
 namespace GraphQL.Conventions.Adapters.Resolvers
 {
-    public class EventStreamResolver : FieldResolver, IEventStreamResolver
+    public class EventStreamResolver : ISourceStreamResolver
```
Comment (Shane32) [inline src/GraphQL.Conventions/Adapters/Resolvers/EventStreamResolver.cs]:
```suggestion
    public class EventStreamResolver : ISourceStreamResolver, IFieldResolver
```

File: src/GraphQL.Conventions/Adapters/Resolvers/EventStreamResolver.cs [hunk 3]
```diff
@@ -1,30 +1,26 @@
...
+        public async ValueTask<IObservable<object>> ResolveAsync(IResolveFieldContext context)
         {
-            var result = base.Resolve(context);
+            var result = await _fieldResolver.ResolveAsync(context);
             if (result is Task<object>)
             {
                 result = (result as Task<object>).Result;
             }
             return (IObservable<object>)result;
         }
```
Comment (Shane32) [inline src/GraphQL.Conventions/Adapters/Resolvers/EventStreamResolver.cs:24]:
```suggestion
        }

        ValueTask<object> IFieldResolver.ResolveAsync(IResolveFieldContext context)
        {
            return new(context.Source);
        }
```

File: src/GraphQL.Conventions/Adapters/GraphTypeAdapter.cs [hunk 4]
```diff
@@ -95,7 +97,7 @@ private FieldType DeriveField(GraphFieldInfo fieldInfo)
                     Type = GetType(fieldInfo.Type),
                     Arguments = new QueryArguments(fieldInfo.Arguments.Where(arg => !arg.IsInjected).Select(DeriveArgument)),
                     Resolver = resolver,
```
Comment (Shane32) [inline src/GraphQL.Conventions/Adapters/GraphTypeAdapter.cs]:
```suggestion
                    Resolver = streamResolver,
```

**Review by Shane32 — CHANGES_REQUESTED** [review]
These three changes should fix subscription support.

File: src/GraphQL.Conventions/Adapters/Resolvers/EventStreamResolver.cs (hunk 1 above)
Comment (pleandre) [reply src/GraphQL.Conventions/Adapters/Resolvers/EventStreamResolver.cs]: Thanks a lot @Shane32

**Comment by pleandre** [comment]
It works with @Shane32 changes, would it be possible to approve this PR if there are no more comments ?
I agree with @sungam3r to change to System Text Json, but it requires to rewrite RequestDeserializer.cs which takes some time. Maybe in another PR.

File: test/GraphQL.Conventions.Tests/Adapters/Engine/Types/CustomJsonGraphType.cs [hunk 5]
```diff
@@ -1,14 +1,18 @@
...
 using GraphQL.Types;
+using GraphQLParser.AST;
 
 // ReSharper disable InconsistentNaming
 
 namespace Tests.Adapters.Engine.Types
 {
     #region .NET Class
-    public class JSON : ValueNode<IDictionary<string, object>>
+    public class JSON : GraphQLValue
```
Comment (Shane32) [inline test/GraphQL.Conventions.Tests/Adapters/Engine/Types/CustomJsonGraphType.cs:10]: FYI this test is completely wrong (not the fault of this PR).  Deserializing a GraphQL request cannot possibly deserialize to a `JSON` node.  Rather, `ParseLiteral` should be parsing a structured tree of objects and values into an `Inputs` object or something, or interpreting a JSON string value stored as a string literal and deserializing it as an `Inputs` object.  Further, the `ParseValue` method should probably just take the object and return it, as the GraphQL serializer should be deserializing variables into a structured map.  It won't ever be a `GraphQLValue` node of any kind.  And `ToAST` should convert the structured map into a tree of nodes.

**Review by Shane32 — APPROVED** [review]
I don't use this project, but I don't see any problems with merging this.  I did read through each change.  @sungam3r Do you want to review?

**Review by tlil — APPROVED** [review]

**Comment by tlil** [comment]
LGTM – will let @sungam3r take a look as well

**Review by sungam3r — APPROVED** [review]
I don't use this project too. Merge as you see fit.

**Pull request merged**